powerauth-java-server/src/test/java/com/wultra/security/powerauth/app/server/service/behavior/tasks/v4/TemporaryKeyBehaviorTest.java

```diff
@@ -160,30 +163,42 @@ void testJwtRequestEcdheValidApplicationScope() throws Exception {
         request.setJwt(jwtRequest);
         final TemporaryPublicKeyResponse response = temporaryKeyBehavior.requestTemporaryKey(request);
         assertNotNull(response.getJwt());
-        final SignedJWT decodedJWT = SignedJWT.parse(response.getJwt());
-        final PublicKey masterPublicKeyEc384 = getMasterPublicKey(defaultVersion);
-
-        assertTrue(temporaryKeyTestService.validateJwtSignature(decodedJWT, masterPublicKeyEc384));
-        assertEquals(defaultVersion.getApplicationKey(), decodedJWT.getJWTClaimsSet().getClaim("applicationKey"));
-        assertEquals(challenge, decodedJWT.getJWTClaimsSet().getClaim("challenge"));
-        assertNull(decodedJWT.getJWTClaimsSet().getClaim("activationId"));
-        assertNotNull(decodedJWT.getJWTClaimsSet().getClaim("sharedSecretResponse"));
+
+        final JWSObjectJSON jws = JWSObjectJSON.parse(response.getJwt());
+        assertEquals(1, jws.getSignatures().size());
+        assertEquals("ES384", jws.getSignatures().get(0).getHeader().getAlgorithm().getName());
```

Comment:
Let's avoid relying on order of signatures, it's extract signature by the algorithm name, I think.